Visitor/User can View a Single Recipe

yasmineezequiel opened the pull request:
[PT-link](https://www.pivotaltracker.com/story/show/169667227) 
## PR Content
In this PR we are adding a feature where the API returns a single recipe.
```
As a Visitor
In order to see recipe details
I would like an API that allows me to view a full recipe
```
## PR Contains:
- Testing of user/visitor can view single article spec.
- Show action for recipe controller.

yasmineezequiel committed:
Adds request spec for single recipe
Generates recipe controller

Commit:
Add test cases to request spec.

tochman commented on app/controllers/recipes_controller.rb:
```diff
@@ -0,0 +1,2 @@
+class RecipesController < ApplicationController
+end
```
I was told that we should use a namespace. Like `V1` This controller should be namespaced if that is correct.

clalil replied:
That is correct - we did agree on the namespace v1.

yasmineezequiel replied:
Changed

yasmineezequiel committed:
Merge branch 'development' of https://github.com/CraftAcademy/foodhub-api into single_recipe

yasmineezequiel committed:
Adds Recipes Controller namespace.

yasmineezequiel committed:
Deletes unneccessary recipe controller.

Commit:
merge pull from Yasmine

yasmineezequiel committed:
Defines show action in recipe controller, adds show action in routes and adds portions in recipe params.

yasmineezequiel committed:
-Adds recipe test response

yasmineezequiel committed:
Refactores test.

yasmineezequiel committed:
Solves merging conflicts.

yasmineezequiel committed:
Adds show action after merge.

Commit:
Merge branch 'single_recipe' of github.com:yasmineezequiel/foodhub-api into single_recipe

Add check for show action in Recipe controller to see if recipe exists.
Add tests to request spec.

yasmineezequiel committed:
Adds http 200 response.

Commit:
coordinate http status codes between request specs

Commit:
Factory creates unique recipe instead of default

Commit:
Remove if statement by using record_not_found

Remove if-statement for show action by utilizing:
  rescue_from ActiveRecord::RecordNotFound

oliverochman commented on app/controllers/v1/recipes_controller.rb:
```diff
@@ -17,9 +17,18 @@ def create
 
+  def show
+    recipe = Recipe.find(params[:id])
+    render json: recipe, serializer: Recipes::IndexSerializer
```
You should use a separate serializer here, please create a `Recipes::ShowSerializer`

Commit:
Add seperate show serializer for show action
